src/main/java/org/usfirst/frc/team4099/robot/subsystems/Arm.kt

```diff
@@ -0,0 +1,208 @@
+
+    enum class MovementState {
+        UP, STATIONARY, DOWN,  HOLD
+    }
+
+    enum class ArmState(val targetPos: Double) {
+        LOW(-7* Math.PI / 6), EXCHANGE(0.0), HIGH(7 * Math.PI / 6), VELOCITY_CONTROL(Double.NaN), STILL(Double.NaN)
+    }
+
+
+
+
+    init {
+        masterSRX.set(ControlMode.MotionMagic, 0.0)
+        slaveSRX1.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        slaveSRX2.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        slaveSRX3.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        masterSRX.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 10)
+        masterSRX.setSensorPhase(false)
+        masterSRX.configNominalOutputForward(+0.0, 0)
+        masterSRX.configPeakOutputForward(+1.0, 0)
+        masterSRX.selectProfileSlot(0, 0)
+        masterSRX.config_kF(0, 0.0, 0)
+        masterSRX.config_kP(0, 0.0, 0)
+        masterSRX.config_kI(0, 0.0, 0)
+        masterSRX.config_kD(0, 0.0, 0)
+        armBaseAngle = ArmConversion.pulsesToRadians(masterSRX.sensorCollection.pulseWidthPosition)
+        brakeTime = Timer.getFPGATimestamp()
+
+    }
+
+
+    /**
+     * Changes power to the arm
+     *
+     * @param power A double for the power going to the arm
+     */
+    fun setArmVelocity(power: Double) {
+        armState = ArmState.VELOCITY_CONTROL
+        masterSRX.set(ControlMode.Velocity, Math.abs(power))
+        armPower = power
+    }
+
+    private fun setArmPosition(position: Double) {
+        armAngle = position
+        masterSRX.set(ControlMode.MotionMagic, ArmConversion.radiansToPulses(position).toDouble())
+    }
+
+    /**
+     * Outputs arm subsystem
+     *
+     */
+    override fun outputToSmartDashboard() {
+        SmartDashboard.putNumber("armPower", armPower)
+    }
+
+    @Synchronized override fun stop() {
+        movementState = MovementState.STATIONARY
+        setArmVelocity(0.0)
+        brake.set(DoubleSolenoid.Value.kReverse)
+    }
+
+    /**
+     * Handles power to arm during each loop cycle
+     *
+     * @constructor Creates loop that controls power to arm during each loop cycle
+     */
+    val loop: Loop = object : Loop {
+
+        override fun onStart() {
+            movementState = MovementState.STATIONARY
+            setArmVelocity(0.0)
+            brake.set(DoubleSolenoid.Value.kForward)
+        }
+
+        /**
+         * Handles events during each loop cycle, depends on arm control mode
+         */
+        override fun onLoop() {
+            synchronized(this@Arm) {
+                // if quadrature velocity is zero, set to hold
+                // after 0.5 seconds of hold, actuate the brake and prevent movement
+                // 0.5 seconds after actuating the brake, set to stationary
+                // once the user tries to move, un-actuate the brake
+                if (masterSRX.sensorCollection.quadratureVelocity == 0) {
```

Comment:
There's both this condition for `quadratureVelocity` as well as the one down at the bottom. Only one is necessary.